daemon/vpn/nordlynx/kernel_space.go

```diff
@@ -10,35 +10,55 @@ import (
 	"sync"
 
 	"github.com/NordSecurity/nordvpn-linux/daemon/vpn"
+	"github.com/NordSecurity/nordvpn-linux/events"
 	"github.com/NordSecurity/nordvpn-linux/internal"
 	"github.com/NordSecurity/nordvpn-linux/tunnel"
 )
 
 type KernelSpace struct {
-	state  vpn.State
-	active bool
-	fwmark uint32
-	tun    *tunnel.Tunnel
+	state           vpn.State
+	active          bool
+	fwmark          uint32
+	tun             *tunnel.Tunnel
+	eventsPublisher *vpn.Events
```

Comment:
Any reason to use *vpn.Events instead of `vpn.InternalVPNPublisher`?

Reply:
`vpn.Events` is the publishing part, vpn implementations use it to publish events, `vpn.InternalVPNPublisher` is the interface that subscribers(`state.StatePublisher`) implement to subscribe to those events.